app/cli.h

```diff
@@ -0,0 +1,206 @@
+    MemoryBundle memoryBundle;
+
+    if (options->get<bool>("bytearray")) {
+      // Load legit values into bytearrays.
+      memoryBundle = getMemoryBundleFromConfig(options);
+    }
+
+    Service service(options, std::move(memoryBundle));
+
+    // Read a large input text blob from stdin
+    std::ostringstream std_input;
+    std_input << std::cin.rdbuf();
+    std::string input = std_input.str();
+
+    ResponseOptions responseOptions;
+    responseOptions.qualityScores = true;
+    responseOptions.alignment = true;
+    responseOptions.alignmentThreshold = 0.2f;
+
+    // Wait on future until Response is complete
+    std::future<Response> responseFuture = service.translate(std::move(input), responseOptions);
+    responseFuture.wait();
+    Response response = responseFuture.get();
+
+    std::cout << "[original]: " << response.source.text << '\n';
+    std::cout << "[translated]: " << response.target.text << '\n';
+    for (int sentenceIdx = 0; sentenceIdx < response.size(); sentenceIdx++) {
+      std::cout << " [src Sentence]: " << response.source.sentence(sentenceIdx) << '\n';
+      std::cout << " [tgt Sentence]: " << response.target.sentence(sentenceIdx) << '\n';
+      std::cout << "Alignments" << '\n';
+      typedef std::pair<size_t, float> Point;
+
+      // Initialize a point vector.
+      std::vector<std::vector<Point>> aggregate(response.source.numWords(sentenceIdx));
+
+      // Handle alignments
+      auto &alignments = response.alignments[sentenceIdx];
+      for (auto &p : alignments) {
+        aggregate[p.src].emplace_back(p.tgt, p.prob);
+      }
+
+      for (size_t src = 0; src < aggregate.size(); src++) {
+        std::cout << response.source.word(sentenceIdx, src) << ": ";
+        for (auto &p : aggregate[src]) {
+          std::cout << response.target.word(sentenceIdx, p.first) << "(" << p.second << ") ";
+        }
+        std::cout << '\n';
+      }
+
+      // Handle quality.
+      auto &quality = response.qualityScores[sentenceIdx];
+      std::cout << "Quality: whole(" << quality.sequence << "), tokens below:" << '\n';
+      size_t wordIdx = 0;
+      bool first = true;
+      for (auto &p : quality.word) {
+        if (first) {
+          first = false;
+        } else {
+          std::cout << " ";
+        }
+        std::cout << response.target.word(sentenceIdx, wordIdx) << "(" << p << ")";
+        wordIdx++;
+      }
+      std::cout << '\n';
+    }
+    std::cout << "--------------------------\n";
+    std::cout << '\n';
+  }
+};
+
+}  // namespace app
+
+/// Template based duplication to call the respective classes based on mode string at run-time.
+template <class CLIAppType>
+void _execute(Ptr<Options> options) {
+  CLIAppType app;
+  app.execute(options);
+}
+
+void execute(const std::string &mode, Ptr<Options> options) {
+  if (mode == "wasm") {
+    _execute<app::WASM>(options);
+  } else if (mode == "native") {
+    _execute<app::Native>(options);
+  } else if (mode == "decoder") {
```

Comment:
Naming here is a bit confusing. Is decoder supposed to be "benchmark-only"

Reply:
decoder is supposed to be same as `marian-decoder`, except using a different translation code-path (the request response one). This is yeah, for our purposes, benchmark only at the moment. 

I can see this collapsing into native somehow. I'll probably turn this into hijacking options and calling service directly, but later.

Reply:
So, decoder = native - extra info besides translation?